shiftcommerce/shift-next
# Feature: PayPal Checkout Integration

## Overview

Github Issue: https://github.com/shiftcommerce/shift-front-end-react/issues/845

This PR contains PayPal integration logic. 

[Part 1: PayPal Checkout Integration - Checkout 1 to 3](https://github.com/shiftcommerce/shift-next/pull/10)
[Part 2: PayPal Checkout Integration - Checkout 3 to 5](https://github.com/shiftcommerce/shift-next/pull/15)

![Payment Method Page](https://user-images.githubusercontent.com/4486874/54922676-8dda5080-4f00-11e9-9450-1eb697c4e13c.png)

## QA

1. checkout branch `feature/790_paypal_checkout_integration`

2. In the `shift-front-end-react` project, checkout branch `issue/790_paypal_checkout_integration`. Boot app (assumption is that you are using yalc)

3. Add items to cart and navigate to the first step of checkout - `payment-method` selection

4. Check that the `Continue Shopping` btn in the cart should be full-width

5. Check that the `Payment Method` step is present and highlighted

6. Check that there are payment method options displayed on the page

7. Check that  clicking the `Pay By Credit/Debit Card` button results in the page navigating to checkout step 2 - `Shipping Address`

8. Check that clicking the `PayPal` button in the page navigates to the PayPal website. 

9. Complete all steps on PayPal and click the continue button

10. When you have been redirected back to the site, check that the shipping address summary is present and that you are redirect to checkout 3 - shipping method. Also ensure that the `Edit` address button is not present and that clicking the shipping address step in the checkout header steps, redirects you back to checkout 1 - Payment Method selection

11. Continue to the review page and place order

![review page](https://user-images.githubusercontent.com/4486874/55413343-14c2a500-5561-11e9-9b72-7df8985bb775.png)

12. Confirm order

![Order confirmation page](https://user-images.githubusercontent.com/4486874/55413401-2906a200-5561-11e9-964b-a140e5a4b2aa.png)

## Deployment
The following environment variables are required for this feature to work:
```
PAYPAL_CLIENT_ID='test'
PAYPAL_CLIENT_SECRET='secret'
ENABLE_PAYPAL_LIVE_ENVIRONMENT=false
```
